code/outils.ss

```diff
@@ -1,37 +1,75 @@
 ;; helpful procedures
 
+;; Convenience for grabbing alist values by key
 (define (lookup key alist)
   (cond ((assoc key alist) => cdr)
-	(else (error 'lookup "key not in alist" key alist))))
+        (else (error 'lookup "key not in alist" key alist))))
 
 (define (lookup-partial key)
   (lambda (alist)
     (lookup key alist)))
```

Comment:
I think we could combine these two functions, ie. there could be default currying without needing extra parentheses if we use a `case-lambda` to join them.  I didn't want to make that change, because I don't think it belongs in this PR.

I added the comment, just becasue I was adding a considerable amount of code relative to the length of the file.

Reply:
Good point. It'll require changing some of the implementations I have for problems, but I think it's a good move!